### ISSUE TITLE
Float serialization to JSON lossy

### ISSUE THREAD

**Issue by mperham** [crystal-lang/crystal#2643]
Serializing a Float to JSON results in data loss:

```cr
require "json"

float = Time.now.epoch_f
p float.to_json # => "1.46413e+09"
```

I'd like the JSON serializer to use "%.6f" or something like that to minimize lossiness.

**Comment by jhass** [comment]
Code is here: https://github.com/crystal-lang/crystal/blob/master/src/json/to_json.cr#L167, culprit is here: https://github.com/crystal-lang/crystal/blob/master/src/float.cr#L195

**Comment by bcardiff** [comment]
Besides monkey patch, a workaround can be done using formatters like http://crystal-lang.org/api/Time/EpochConverter.html

**Comment by mperham** [comment]
Thanks @jhass, #2644 submitted.

@bcardiff That doesn't work for object.to_json, only for JSON.mapping, afaik.

**Comment by asterite** [comment]
I added an answer to this here: https://github.com/crystal-lang/crystal/pull/2644#issuecomment-221611181

For preserving float digits you can probably use `BigFloat` until we enhance `Float#to_s`

**Comment by mperham** [comment]
Thanks, I might be able to get that to work!

```
float = Time.now.epoch_f
p float.to_s
p BigFloat.new(float).to_s

"1.46419e+09"
"1464191446.0133600235"
```

**Comment by asterite** [comment]
This is now fixed in 0.17.4:

```crystal
require "json"

float = Time.now.epoch_f
puts float.to_json # => 1464289272.8178599
```
